python/rikai/types/vision.py

```diff
@@ -104,6 +105,20 @@ def from_pil(img: PILImage, uri: Union[str, Path]) -> Image:
                 copy(fobj.name, uri)
         return Image(uri)
 
+    @staticmethod
+    def from_bytes(img: bytes, uri: Union[str, Path]) -> Image:
+        """Create an image in memory from a byte string.
+
+        Parameters
+        ----------
+        img : bytes
+            An PIL Image instance
+        uri : str or Path
+            The URI to store the image externally.
+        """
+        img = PILImage.open(BytesIO(img))
```

Comment:
if this is returning an Image then you gotta close your resources.

```
with PILImage.open(BytesIO(img_bytes)) as pil_img:
    return self.from_pil(pil_img, uri)
```

Reply:
good catch, thanks!